src/main/java/com/cognifide/aemrules/htl/lex/DoctypeTokenizer.java

```diff
@@ -0,0 +1,66 @@
+/*-
+ * #%L
+ * AEM Rules for SonarQube
+ * %%
+ * Copyright (C) 2015-2018 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.aemrules.htl.lex;
+
+import java.io.IOException;
+import java.io.StreamTokenizer;
+import java.io.StringReader;
+import java.util.List;
+import org.sonar.plugins.html.node.Attribute;
+import org.sonar.plugins.html.node.DirectiveNode;
+import org.sonar.plugins.html.node.Node;
+
+class DoctypeTokenizer extends AbstractTokenizer<List<Node>> {
+
+    public DoctypeTokenizer(String startToken, String endToken) {
+        super(startToken, endToken);
+    }
+
+    private static void parseToken(DirectiveNode node) {
+        String code = node.getCode();
+        StreamTokenizer tokenizer = new StreamTokenizer(new StringReader(code));
+        tokenizer.quoteChar('"');
+        try {
+            while (tokenizer.nextToken() != StreamTokenizer.TT_EOF) {
+                if (tokenizer.sval != null) {
```

Comment:
Maybe
`if(tokenizer.sval != null && node.getNodeName() == null) {
} else {
}`

Reply:
Im not so sure about that change. It will l change the logic.

Reply:
Yes you are right. Sorry for misleading comment